plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityAngry.java

```diff
@@ -5,87 +5,64 @@
-                || (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18) && entity instanceof Vindicator);
-    }
-
-    public static EntityAngry getFrom(ObjectTag entity) {
-        if (!describes(entity)) {
-            return null;
-        }
-        else {
-            return new EntityAngry((EntityTag) entity);
-        }
-    }
+    // <--[property]
+    // @object EntityTag
+    // @name angry
+    // @input ElementTag(Boolean)
+    // @description
+    // Controls whether an entity is angry.
+    // -->
 
-    public static final String[] handledMechs = new String[] {
-            "angry"
-    };
-
-    public EntityAngry(EntityTag entity) {
-        this.entity = entity;
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof Wolf
+                || entity.getBukkitEntity() instanceof PigZombie
+                || (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_18) && entity.getBukkitEntity() instanceof Vindicator);
     }
 
     EntityTag entity;
 
     @Override
-    public String getPropertyString() {
-        if (isWolf()) {
-            return getWolf().isAngry() ? "true" : null;
+    public ElementTag getPropertyValue() {
+        if (isWolf() && getWolf().isAngry()) {
+            return new ElementTag(true);
         }
-        else if (isPigZombie()) {
-            return getPigZombie().isAngry() ? "true" : null;
+        else if (isPigZombie() && getPigZombie().isAngry()) {
+            return new ElementTag(true);
         }
-        else if (isVindicator()) {
-            return getVindicator().isJohnny() ? "true" : null;
+        else if (isVindicator() && getVindicator().isJohnny()) {
+            return new ElementTag(true);
         }
         return null;
     }
 
+    @Override
+    public void setPropertyValue(ElementTag param, Mechanism mechanism) {
+        if (mechanism.requireBoolean()) {
+            if (isWolf()) {
+                getWolf().setAngry(param.asBoolean());
+            }
+            else if (isPigZombie()) {
+                getPigZombie().setAngry(param.asBoolean());
+            }
+            else if (isVindicator()) {
+                getVindicator().setJohnny(param.asBoolean());
+            }
+        }
+    }
+
     @Override
     public String getPropertyId() {
         return "angry";
     }
 
     public static void register() {
-
-        // <--[tag]
-        // @attribute <EntityTag.angry>
-        // @returns ElementTag(Boolean)
-        // @mechanism EntityTag.angry
-        // @group properties
-        // @description
-        // If the entity is a wolf or PigZombie, returns whether the entity is angry.
-        // If the entity is a Vindicator, returns whether it is in "Johnny" mode.
-        // -->
-        PropertyParser.registerTag(EntityAngry.class, ElementTag.class, "angry", (attribute, entity) -> {
-            if (entity.isWolf()) {
-                return new ElementTag(entity.getWolf().isAngry());

```

Comment:
did you test that the tag still works exactly the same as before (spoiler: it doesn't, you changed it)